src/Application/Actions/Person/Update.php

```diff
@@ -200,7 +204,10 @@ protected function action(Request $request, array $args): Response
 
         $this->stripeClient->customers->update($person->stripe_customer_id, $customerDetails);
 
-        if (!$personHadPassword && $person->getPasswordHash() !== null) {
+        if (!$personHadPassword && $personHasPasswordNow) {
+            $customerDetails['metadata']['hasPasswordSince'] = (new \DateTimeImmutable())->format('Y-m-d H:i:s');
```

Comment:
Could we add a comment about what `hasPasswordSince` is used for? I realise it will be used for the current work in matchbot but that might not be obvious if we look at this in six month's time.